lib/api/src/externals/function.rs

```diff
@@ -283,6 +304,11 @@ impl Function {
             host_env: Box::into_raw(box_env) as *mut _,
         };
         let signature = function.ty();
+        // TODO: look into removing transmute by changing API type signatures
+        let function_ptr = Some(std::mem::transmute::<
+            fn(_, _) -> Result<(), _>,
+            fn(_, _) -> Result<(), _>,
+        >(Env::init_with_instance));
```

Comment:
I'm not super sure why this is needed. Can you provide some context @MarkMcCaskey ?

Reply:
Which part? The function_ptr at all or the transmute?

We need the function pointer to initialize the data, so far this is the most direct place to put it so we can access where we need it.

We have to transmute it because `WasmerEnv` isn't known to the `vm` so we can't use the real name of the type, I have some work in progress trying to make it so vm can see `WasmerEnv` but it's a bit stuck